Add details on the "Discover relation" entry (#5915)
qgis/QGIS-Documentation: docs/user_manual/working_with_vector/attribute_table.rst

providers (so you can also use it with shape and csv files) and all you have
to do is to tell QGIS the relations between your tables.

Defining 1-N relations
......................

The first thing we are going to do is to let QGIS know about the relations
between the layers. This is done in :menuselection:`Project --> Properties...`.
Open the :guilabel:`Relations` tab and click on |signPlus| :guilabel:`Add Relation`.

* **Name** is going to be used as a title. It should be a human readable string,
  describing, what the relation is used for. We will just call say **airport_relation**
  in this case.
* **Referenced Layer (Parent)** also considered as parent layer, is the one with

>   the primary key, pointed to, so here it is the ``regions`` layer. You can define

Reviewer: ```suggestion
  the primary key, pointed to, so here it is the ``regions`` layer. You need to define
```

  the primary key of the referenced layer, so it is ``ID``.
* **Referencing Layer (Child)** also considered as child layer, is the one with
  the foreign key field on it. In our case, this is the ``airports`` layer. For
  this layer you need to add a referencing field which points to the other
  layer, so this is ``fk_region``.

  .. note:: Sometimes, you need more than a single field to uniquely identify
   features in a layer. Creating a relation with such a layer requires

>    a **composite key**, ie more than a single pair of referencing/referenced

Reviewer: ```suggestion
   a **composite key**, ie more than a single pair of matching
```

>    fields. Use the |signPlus| :sup:`Add fields to pair` button

Author: ```suggestion
   fields. Use the |signPlus| :sup:`Add new field pair as part of a composite
```

>    to add as many pairs as necessary.

Author: ```suggestion
   foreign key` button to add as many pairs as necessary.
```

.. figure:: img/relations2.png
   :align: center

   Adding a relation between regions and airports layers

From the :guilabel:`Relations` tab, you can also press the |signPlus|
:guilabel:`Discover Relation` button to fetch the relations available from

> the providers of the current layers. This is possible for layers stored in

Reviewer: What are "current layers"?

Author: ```suggestion
the providers of the loaded layers. This is possible for layers stored in
```
?

Reviewer: :+1:

> data providers like PostgreSQL or SpatiaLite.

Reviewer: Doesn't it also work with GeoPackage?

Author: Good question, and the reason why I wanted a dev around ;). I found a discover_relation in both postgresql, spatialite (and ogr - I don't know which format) classes: https://github.com/qgis/QGIS/search?q=discoverrelations&type=Code. Moreover, in https://docs.qgis.org/3.10/en/docs/user_manual/working_with_vector/attribute_table.html#introducing-many-to-many-n-m-relations it's mentioned that the "discover relation" can be used only for PG and not gpkg (obsolete information or still true? I don't know and can't reasonably check myself)

Reviewer: Where did you find it in ogr? That would hint to gpkg but I didn't see it in the search results (which would be a hint it's not supported)

Author: Oups. Sorry. I misread vectorprovider as ogr.

Reviewer: In this case, let's leave it as is. Possibly open a feature request for gopackage support?

Author: Feature request opened at https://github.com/qgis/QGIS/issues/38013
